deps/ledger-zxlib/src/sigutils.c

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+*   (c) 2020 Zondax GmbH
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+********************************************************************************/
+
+#include <sigutils.h>
+#include <zxmacros.h>
+
+err_convert_e convertDERtoRSV(const uint8_t *inSignatureDER,
+                              unsigned int inInfo,
+                              uint8_t *outR,
+                              uint8_t *outS,
+                              uint8_t *outV) {
+
+    // https://github.com/libbitcoin/libbitcoin-system/wiki/ECDSA-and-DER-Signatures#serialised-der-signature-sequence
+    // 0                [1 byte]   - DER Prefix
+    // 1                [1 byte]   - Payload len
+    // 2                [1 byte]   - R Marker. Always 02
+    // 3                [1 byte]   - R Len                      RLEN
+    // ROFFSET ...      [.?. byte] - R                          ROFFSET
+    // ROFFSET+RLEN     [1 byte]   - S Marker. Always 02
+    // ROFFSET+RLEN+1   [1 byte]   - S Length                   SLEN
+    // ROFFSET+RLEN+2   [.?. byte] - S                          SOFFSET
+    // Prepare response
+    // R [32]
+    // S [32]
+    // V [1]
+
+    const uint8_t derPrefix = *(inSignatureDER);
+    if (derPrefix != 0x30) {
+        return invalid_derPrefix;
+    }
+
+    const uint8_t payloadLen = *(inSignatureDER + 1);
+    if (payloadLen < 4 + 32 + 2 + 32 || payloadLen > 4 + 33 + 2 + 33) {
+//        return invalid_payloadLen;
```

Comment:
This one escaped again! :)
```suggestion
        return invalid_payloadLen;
```